internal/controller/sync.go

```diff
@@ -560,14 +560,16 @@ func (r *RouteController) populateRoute(ctx context.Context, route *routev1.Rout
 	// final Sanity checks
 	var key crypto.Signer
 
-	// get private key and signed certificate from Secret
+	// get private key, signed certificate and ca chain certficates from Secret
 	k, err := utilpki.DecodePrivateKeyBytes(secret.Data["tls.key"])
 	if err != nil {
 		return err
 	}
 	key = k
 
-	certificate, err := utilpki.DecodeX509CertificateBytes(secret.Data["tls.crt"])
+	certificates, err := utilpki.DecodeX509CertificateSetBytes(secret.Data["tls.crt"])
+
+	certificate := certificates[0]
```

Comment:
this is dangerous imho - since there is no guarantee - or at least i do not see one - that the concrete cert is the first one in the chain

Reply:
hum

This is exactly what the previously used function `utilpki.DecodeX509CertificateBytes` does: return the first element of the array from the `utilpki.DecodeX509CertificateSetBytes` function.

see : https://github.com/cert-manager/cert-manager/blob/63f83c901659f807f9f6c49092e26d24a7b527bd/pkg/util/pki/parse.go#L118

I just moved the extraction in this function so I don't have to produce a pull request on the project `cert-manager`.